src/operator/random/sample_op.h

```diff
@@ -693,10 +750,12 @@ inline bool SampleOpType(const nnvm::NodeAttrs& attrs,
     }
   }
   bool dtype_ok = (dtype == kFloat16) || (dtype == kFloat32) ||
```

Comment:
Do we correctly throw an exception if users call randint(..,.., dtype='float32')?

Reply:
updated to handle randint (int op type) separate from other rand functions